nixos/modules/security/doas.nix

```diff
@@ -43,34 +41,42 @@ let
           (map (grp: "permit ${opts} ${mkGrpString grp} ${as} ${cmd} ${args}") rule.groups)
       )
     ];
+
+    sudoShimOptions = {
+      options = {
+        enable = mkEnableOption (mdDoc "sudo shim that uses doas");
+        package = mkPackageOptionMD pkgs "doas-sudo-shim" {};
+      };
+    };
 in
 {
 
   ###### interface
 
   options.security.doas = {
 
-    enable = mkOption {
-      type = with types; bool;
-      default = false;
-      description = lib.mdDoc ''
-        Whether to enable the {command}`doas` command, which allows
-        non-root users to execute commands as root.
-      '';
-    };
+    enable = mkEnableOption (mdDoc "{command}`doas` command, which allows non-root users to excute commands as root");
+
+    package = mkPackageOptionMD pkgs "doas" {};
 
     wheelNeedsPassword = mkOption {
       type = with types; bool;
       default = true;
-      description = lib.mdDoc ''
+      description = mdDoc ''
         Whether users of the `wheel` group must provide a password to
         run commands as super user via {command}`doas`.
       '';
     };
 
+    sudoShim = mkOption {
```

Comment:
Why is this a submodule instead of just being nested options?

Reply:
I used submodule because technically it is by itself a completely different package, with its own package option and enable option. Which has nothing to do with doas project, only depending on it. A bit like `virtualisation.libvirtd.qemu`.

Reply:
Don't do that, just use nested options. Submodules are mostly useful combined with lists or sets of arbitrary keys, e.g. systemd.services.*. It doesn't add anything here and on top of that submodules also come with performance implications due to the nested context.

Reply:
Changed to nested options